Packs/DigitalGuardian/Integrations/DigitalGuardianARCEventCollector/DigitalGuardianARCEventCollector.py

```diff
@@ -22,239 +24,246 @@ class Client(BaseClient):
+    limit = arg_to_number(args.get('limit')) or 1000
+    events, last_run = fetch_events(client, limit=limit)
+
+    event_list = list(events)
+    if event_list:
+        human_readable = tableToMarkdown(
+            name='Test Events',
+            t=[{key: value for key, value in event.items() if value != "-"} for event in event_list],
+        )
+    else:
+        human_readable = 'No events found.'
+
+    demisto.debug(f'Displayed limit of {limit} events from response')
 
-    # Save the next_run as a dict with the last_fetch key to be stored
-    next_run = {'start_time': time_of_event, 'id_list': id_list}
-    demisto.debug(f'Setting next run {next_run}.')
-    return next_run, event_list_for_push
+    return event_list, last_run, CommandResults(readable_output=human_readable)
 
 
-def add_time_to_events(events: list[dict]) -> None:
+def push_and_set_last_run(client: Client, events: Iterable, last_run: dict) -> None:
     """
-    Adds the _time key to the events.
+    Saves last run and moves internal bookmark in the API for the time fetch-events is invoked.
+
     Args:
-        events: List[Dict] - list of events to add the _time key to.
-    Returns:
-        list: The events with the _time key.
+        client (Client): Digital Guardian client.
+        events (Iterable): Events (A generator object if internal fetch or list of from get-events command).
+        last_run (dict): Dictionary with 'bookmark_values' and 'search_after_values' from raw API response.
     """
-    if events:
-        for event in events:
-            create_time = arg_to_datetime(arg=event.get('inc_mtime')) if event.get('inc_mtime') else None
-            event['_time'] = create_time.strftime(DATE_FORMAT) if create_time else None
+    send_events_to_xsiam(events=events, vendor=VENDOR, product=PRODUCT)
+
+    demisto.debug(f'Setting export bookmark after run: {last_run}.')
+    client.set_export_bookmark()  # API-managed bookmark / pointer removes the need for setting and getting last run on our end
 
 
 def main() -> None:  # pragma: no cover
     params = demisto.params()
     args = demisto.args()
     command = demisto.command()
-    auth_url = params.get('auth_server_url')
-    gateway_url = params.get('gateway_base_url')
+
+    auth_url = params.get('auth_server_url', '')
+    base_url = params.get('gateway_base_url', '')
     client_id = params.get('credentials', {}).get('identifier', '')
     client_secret = params.get('credentials', {}).get('password', '')
-    export_profile = params.get('export_profile')
+    export_profile = params.get('export_profile', '')
     verify_certificate = not params.get('insecure', False)
-    base_url = urljoin(gateway_url, f'/rest/2.0/export_profiles/{export_profile}/export')
-    demisto.debug(f'the base url is:{base_url}')
-    next_run = None
+
+    demisto.debug(f'Running {VENDOR} event collector with base url: {base_url}')
```

Comment:
```suggestion
    demisto.debug(f'{base_url=}')
```
[see here](https://panw-global.slack.com/archives/G011E63JXPB/p1730217052505469)

Reply:
Fixed in commit 582763c169821e56b0a063bc42e00239eb151efc